dwst/scripts/particles.js

```diff
@@ -22,6 +22,21 @@ const specialChars = [
   '\\',
 ];
 
+const legalInstructionNameChars = [
+  'A', 'B', 'C', 'D', 'E',
+  'F', 'G', 'H', 'I', 'J',
+  'K', 'L', 'M', 'N', 'O',
+  'P', 'Q', 'R', 'S', 'T',
+  'U', 'V', 'W', 'X', 'Y',
+  'Z', 'a', 'b', 'c', 'd',
+  'e', 'f', 'g', 'h', 'i',
+  'j', 'k', 'l', 'm', 'n',
+  'o', 'p', 'q', 'r', 's',
+  't', 'u', 'v', 'w', 'x',
+  'y', 'z', '_',
```

Comment:
I don't think we have any instructions with `snake_case` at the moment. Actually we don't have any instructions with `camelCase´ either. Perhaps we should only allow lower case letters. We could always add support for snake case or camel case later. We should probably only add support for one or the other. I think we should postpone the decision and only support lower case for the time being.

Reply:
Also, I find this hard to read. Maybe do something like.
```
const aCode = 'a'.charCodeAt(0);
const zCode = 'z'.charCodeAt(0);
const charCodes = range(aCode, zCode + 1);
const chars = String.fromCharCode(charCodes)
```

Reply:
My example above creates a string with characters from a to z. You could do split('') to turn it into an array. Or you could make readUntil and readWhile accept a string argument and call join('') on the specialChars array.
```
const specialChars = [
  '$',
  '\\',
].join('');
```

Reply:
`range` is available from utils.js

Reply:
One more options is
```
const chars = charCodes.map(charCode => String.fromCharCode(charCode));
```

Reply:
Diddly.

Reply:
So this works fine when your run the tests but fails in Chrome?

Reply:
What does eslint think about your import?

Reply:
Anyway, you are importing it wrong. Utils exports an object. You need to do.
```
import utils from '../utils.js';
utils.range( );
```
At least at the moment it works like that. We could of course change it if you don't like how it works.

Reply:
I think I ended up doing it this way so it would be easy to see in context what calls are part of the utils package.